keps/sig-scheduling/5004-dra-extended-resource/README.md

```diff
@@ -0,0 +1,649 @@
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Extended resource provides a simple, concise approach to describe resource
+capacity, and resource consumption. In constrast, Dynamic Resource
+Allocation (DRA) provides more expressive, flexible, powerful, yet more
+complicated, and harder to use approach.
+
+This KEP provides a solution to enable cluster administrators and device drivers
+to advertise the resources as extended resource, and enables the application
+developers, and operators to continue using extended resource to request for such
+resources.
+
+## Motivation
+
+There are three major motivations for the solution in this KEP.
+
+* Enable the existing applications to run without modification.
+
+* Enable application developers, operators to transition to DRA gradually at
+  their own pace.
+
+* Enable cluster administrators to transition to DRA gradually at their own pace.
+
+For example, the following `Deployment` can be installed without modification on a
+cluster with DRA `ResourceSlice` below.
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: demo
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: demo
+  template:
+    metadata:
+      labels:
+        app: demo
+    spec:
+      containers:
+      - name: demo
+        image: nvidia/cuda:8.0-runtime
+        command: ["/bin/sh", "-c"]
+        args: ["nvidia-smi && tail -f /dev/null"]
+        resources:
+          limits:
+            nvidia.com/gpu: 1
+```
+
+```yaml
+apiVersion: resource.k8s.io/v1beta1
+kind: ResourceSlice
+metadata:
+  name: gke-drabeta-n1-standard-4-2xt4-346fe653-zrw2-gpu.nvidia.coqj92d
+spec:
+  devices:
+  - basic:
+    name: gpu-0
+  - basic:
+    name: gpu-1
+  - basic:
+    name: gpu-2
+  - basic:
+    name: gpu-3
+  - basic:
+    name: gpu-4
+  - basic:
+    name: gpu-5
+  - basic:
+    name: gpu-6
+  - basic:
+    name: gpu-7
+  driver: gpu.nvidia.com
+  nodeName: gke-drabeta-n1-standard-4-2xt4-346fe653-zrw2
+```
+
+With this motivating example in mind. We define the following goals and
+non-goals of this KEP.
+
+### Goals
+
+* Introduce the ability for DRA to advertise extended resources listed in a
+  ResourceSlice, and for the scheduler to consider them for allocation.
+
+* Enable application operators to use the existing extended resource limits to
+request for DRA resources.
+
+### Non-Goals
+
+* Extended resource support is not added just for easing the transition to DRA
```

Comment:
This sounds more like an additional goal than a non-goal.

Reply:
right, moved it to Goals, and added a Non-Goals.